apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/MoveBlockButtons/MoveBlockButtons.tsx

```diff
@@ -0,0 +1,145 @@
+import { ReactElement, useEffect } from 'react'
+import { styled } from '@mui/material/styles'
+import Box from '@mui/material/Box'
+import Button from '@mui/material/Button'
+import ButtonGroup from '@mui/material/ButtonGroup'
+import Typography from '@mui/material/Typography'
+import KeyboardArrowUpRoundedIcon from '@mui/icons-material/KeyboardArrowUpRounded'
+import KeyboardArrowDownRoundedIcon from '@mui/icons-material/KeyboardArrowDownRounded'
+import { gql, useMutation } from '@apollo/client'
+import { TreeBlock, searchBlocks } from '@core/journeys/ui'
+import { BlockOrderUpdate } from '../../../../../../__generated__/BlockOrderUpdate'
+import { useJourney } from '../../../../../libs/context'
+
+const StyledMoveButton = styled(Button)(({ theme }) => ({
+  backgroundColor: theme.palette.background.paper,
+  color: theme.palette.secondary.light,
+  border: `1px solid ${theme.palette.divider}`,
+  '&:first-of-type': {
+    borderTopLeftRadius: 12,
+    borderBottomLeftRadius: 12
+  },
+  '&:last-of-type': {
+    borderTopRightRadius: 12,
+    borderBottomRightRadius: 12
+  },
+  '&.MuiButtonGroup-grouped:not(:last-of-type)': {
+    borderRight: 'none',
+    borderColor: theme.palette.divider
+  },
+  '&.MuiButtonGroup-grouped:not(:last-of-type).Mui-disabled': {
+    borderColor: theme.palette.divider
+  },
+  '&:hover': {
+    backgroundColor: theme.palette.divider
+  },
+  height: 60
+}))
+
+export const BLOCK_ORDER_UPDATE = gql`
+  mutation BlockOrderUpdate($id: ID!, $journeyId: ID!, $parentOrder: Int!) {
+    blockOrderUpdate(
+      id: $id
+      journeyId: $journeyId
+      parentOrder: $parentOrder
+    ) {
+      id
+      parentOrder
+    }
+  }
+`
+interface MoveBlockButtonsProps {
+  selectedBlock: TreeBlock
+  selectedStep: TreeBlock
+}
+
+export function MoveBlockButtons({
+  selectedBlock,
+  selectedStep
+}: MoveBlockButtonsProps): ReactElement {
+  const [blockOrderUpdate] = useMutation<BlockOrderUpdate>(BLOCK_ORDER_UPDATE)
+  const journey = useJourney()
+
+  const parentBlock =
+    selectedBlock.parentBlockId != null
+      ? searchBlocks(selectedStep.children, selectedBlock.parentBlockId)
+      : selectedStep
+
+  const coverBlock =
+    parentBlock != null &&
+    parentBlock.__typename === 'CardBlock' &&
+    parentBlock.coverBlockId != null
+      ? searchBlocks(parentBlock.children, parentBlock.coverBlockId)
+      : undefined
+
+  useEffect(() => {
+    const fixCoverOrder = async (): Promise<void> => {
+      if (coverBlock != null && coverBlock.parentOrder !== 0) {
+        await blockOrderUpdate({
+          variables: {
+            id: coverBlock.id,
+            journeyId: journey.id,
+            parentOrder: 0
+          }
+        })
+      }
+    }
+
+    void fixCoverOrder()
+  }, [coverBlock, journey, blockOrderUpdate])
+
+  const handleMove = async (move: 'up' | 'down'): Promise<void> => {
```

Comment:
could you just provide a function that returns a function?
```suggestion
  const handleMove = async (move: 'up' | 'down'): () => Promise<void> => {
```

would simplify button logic below
```
          onClick={handleMove('up')}
```

Reply:
Every async function must return a Promise typescript error - it suggests `Promise<() => Promise<void>>` which isn't right either..

Reply:
Every async function must return a Promise typescript error - it suggests `Promise<() => Promise<void>>` which isn't right either..

Reply:
Every async function must return a Promise typescript error - it suggests `Promise<() => Promise<void>>` which isn't right either..

Reply:
Every async function must return a Promise typescript error - it suggests `Promise<() => Promise<void>>` which isn't right either.. but noted to simplify onClick